dlup/_image.py

```diff
@@ -220,24 +286,17 @@ def read_region(
         resulting tile size of ``(tile_size, tile_size)`` with a scaling factor of 0.5, we can use:
         >>>  wsi.read_region(location=(coordinate_x, coordinate_y), scaling=0.5, size=(tile_size, tile_size))
         """
-        owsi = self._wsi
+        wsi = self._wsi
         location = np.asarray(location)
         size = np.asarray(size)
         level_size = np.array(self.get_scaled_size(scaling))
 
-        if (size < 0).any():
-            raise ValueError(f"Size values must be greater than zero. Got {size}")
-
-        if ((location < 0) | ((location + size) > level_size)).any():
-            raise ValueError(
-                f"Requested region is outside level boundaries. "
-                f"{location.tolist()} + {size.tolist()} (={(location + size).tolist()}) > {level_size.tolist()}."
-            )
+        _check_size_and_location(location, size, level_size)
 
-        native_level = owsi.get_best_level_for_downsample(1 / scaling)
-        native_level_size = owsi.level_dimensions[native_level]
-        native_level_downsample = owsi.level_downsamples[native_level]
-        native_scaling = scaling * owsi.level_downsamples[native_level]
+        native_level = wsi.get_best_level_for_downsample(1 / scaling)
+        native_level_size = wsi.level_dimensions[native_level]
+        native_level_downsample = wsi.level_downsamples[native_level]
+        native_scaling = scaling * wsi.level_downsamples[native_level]
```

Comment:
native_scaling = scaling * wsi.level_downsamples[native_level]  --> could also be 
 native_scaling = scaling * native_level_downsample